lib/models/device.coffee

```diff
@@ -27,6 +28,38 @@ configModel = require('./config')
 applicationModel = require('./application')
 auth = require('../auth')
 
+MIN_SUPERVISOR_APPS_API = '1.7.1'
+
+###*
+# @summary Check supervisor version compatibility using semver
+# @name isCompatible
+# @orivate
+# @function
+# @memberof resin.models.device
+#
+# @param {String} version - version under check
+# @param {String} minVersion - minimum accepted version
+# @fulfil {Boolean} - is compatible
+# @reject {Error} Will reject if the given version is < than the given minimum version
+# @returns {Promise}
+#
+# @example
+# resin.models.device.isCompatible(version, MIN_VERSION).then(function(isCompatible) {
+# 	console.log(isCompatible);
+# });
+#
+# @example
+# resin.models.device.isCompatible(version, MIN_VERSION, function(error, isCompatible) {
+# 	if (error) throw error;
+# 	console.log(isCompatible);
+# });
+###
+isCompatible = (version, minVersion) ->
```

Comment:
This would be nicer (imo) as:
```coffee
isCompatible = Promise.method (version, minVersion) ->
	if semver.lt(version, minVersion)
		throw new Error("Incompatible supervisor version: #{version} - must be >= #{minVersion}")
	return true
```